boot/flags.go

```diff
@@ -334,6 +343,16 @@ func HostUbuntuDataForMode(mode string) ([]string, error) {
 		if exists, _, _ := osutil.DirExists(installModeLocation); exists {
 			runDataRootfsMountLocations = []string{installModeLocation}
 		}
+
+	case ModeFactoryReset:
+		// In factory reset, our conditions are a lot similar to install mode,
+		// as we recreate the ubuntu-data partition. Make similar assumptions
+		// and checks like ModeInstall. We also make sure to check as we don't even
+		// know if ubuntu-data has been mounted yet.
```

Comment:
Nitpick (assuming I'm not misunderstanding):
```suggestion
		// and checks like ModeInstall. We also make sure to check whether
		// ubuntu-data has been already mounted.
```

There's something scary in this comment though: it sounds like the behaviour here is different depending on whether ubuntu-data has been mounted or not. It might be worth adding a comment to explain what is the expected case (if one case is more likely than the other), and why both cases can happen and how we are dealing with them.

Reply:
Ah ok, I hadn't realized this comment was ill-formed - I'll reword it. I wanted to point out that ubuntu-data might not be mounted as snap-bootstrap does not mount it. Thank you for pointing this out!